tests/vppinputasync.cpp

```diff
@@ -15,22 +15,28 @@
  */
 #include "vppinputasync.h"
 
+static const pthread_t INVALID_THREAD_ID = ((pthread_t)-1);
+
 VppInputAsync::VppInputAsync()
-    :m_cond(m_lock), m_eos(false),
-     m_queueSize(0),m_quit(false)
+    : m_cond(m_lock)
+    , m_eos(false)
+    , m_queueSize(0)
+    , m_startupSize(0)
+    , m_thread(INVALID_THREAD_ID)
+    , m_quit(false)
 {
 }
 
 
 SharedPtr<VppInput>
-VppInputAsync::create(const SharedPtr<VppInput>& input, uint32_t queueSize)
+VppInputAsync::create(const SharedPtr<VppInput>& input, uint32_t queueSize, uint32_t startupSize)
 {
     SharedPtr<VppInput> ret;
 
     if (!input || !queueSize)
         return ret;
     SharedPtr<VppInputAsync> async(new VppInputAsync());
-    if (!async->init(input, queueSize)) {
+    if (!async->init(input, queueSize, startupSize)) {
         ERROR("init VppInputAsync failed");
         return ret;
```

Comment:
I see, that won't work due to mismatched types.  I suppose you can instead set ret=false to handle the failure.